sdk/inc/azure/core/az_result.h

```diff
@@ -80,8 +80,12 @@ typedef enum
   AZ_ERROR_NOT_SUPPORTED = _az_RESULT_MAKE_ERROR(_az_FACILITY_CORE, 7),
 
   /// An external dependency required to perform the operation was not provided. The operation needs
-  /// an implementation of the platform layer or an HTTP transport adapter.
-  AZ_ERROR_DEPENDENCY_NOT_PROVIDED = _az_RESULT_MAKE_ERROR(_az_FACILITY_CORE, 8),
+  /// an implementation of HTTP transport adapter.
+  AZ_ERROR_DEPENDENCY_NOT_PROVIDED_HTTP = _az_RESULT_MAKE_ERROR(_az_FACILITY_CORE, 8),
+
+  /// An external dependency required to perform the operation was not provided. The operation needs
+  /// an implementation of the platform layer.
+  AZ_ERROR_DEPENDENCY_NOT_PROVIDED_PAL = _az_RESULT_MAKE_ERROR(_az_FACILITY_CORE, 9),
```

Comment:
Also, for readability, should they be AZ_ERROR_ HTTP_ADAPTER/PLATFORM _DEPENDENCY_NOT_PROVIDED? I understand the way you put it is better in terms of sorting and therefore grouping, but for the reader, it could be better when done the other way.